### PR TITLE
Import cached binary tables: fail for inconsistent chunk sizes

### PR DESCRIPTION
**Pull request by mweisgut**
With this PR, the table generator fails if an imported cached binary table has a different chunk size than the one defined in the benchmark configuration.

Additionally, this PR introduces minor format changes in `compare_benchmarks.py`, performed by our format script.

### CODE REVIEW

**Comment by mrks** [comment]
> Additionally, this PR introduces minor format changes in compare_benchmarks.py, performed by our format script.

Why didn't this happen automatically? @Bouncner, did you call format? Not trying to assign any blame, just trying to understand.

File: src/benchmarklib/abstract_table_generator.cpp [hunk 1]
```diff
@@ -176,8 +176,8 @@ void AbstractTableGenerator::generate_and_store() {
     for (auto& [table_name, table_info] : table_info_by_name) {
       const auto& table = table_info.table;
       if (table->chunk_count() > 1 && table->get_chunk(ChunkID{0})->size() != _benchmark_config->chunk_size) {
-        std::cout << "- WARNING: " << table_name << " was loaded from binary, but has a mismatching chunk size of "
-                  << table->get_chunk(ChunkID{0})->size() << std::endl;
+        Fail("Table '" + table_name + "' was loaded from binary, but has a mismatching chunk size of " +
+             std::to_string(table->get_chunk(ChunkID{0})->size()));
```
Comment (mrks) [inline src/benchmarklib/abstract_table_generator.cpp:180]: "Delete cached files or use --dont_cache_binary_tables"?

**Comment by Bouncner** [comment]
>Why didn't this happen automatically? @Bouncner, did you call format? Not trying to assign any blame, just trying to understand.

I did happen automatically and I removed this space to get the CI-lint running. @mweisgut : which check made you add this space, the formatter? Looks like the formatter adds it and the linter is then complaining about it.

**Comment by Bouncner** [comment]
This is a known issue. See the #2200 and #2201.

**Comment by mweisgut** [comment]
> > Why didn't this happen automatically? @Bouncner, did you call format? Not trying to assign any blame, just trying to understand.
> 
> I did happen automatically and I removed this space to get the CI-lint running. @mweisgut : which check made you add this space, the formatter? Looks like the formatter adds it and the linter is then complaining about it. Can you please double check?

Yes, the space was added by the formatter and the linter complains about it.

Linter: `scripts/compare_benchmarks.py:301:104: E203 whitespace before ':'`

**Review by mrks — APPROVED** [review]

**Review by julianmenzler — COMMENTED** [review]
In addition to the SF, we could create subfolders for different SF/Chunk-Size variations. For example:
/tpch_cached_tables/sf-0.010000_chunksize-1000

 Might be worth a thought for larger test series.

**Comment by mrks** [comment]
No strong opinion about that. That might lead to a lot of unused data flying around. An alternative would be to call the binary from different folders, e.g., `hyrise/build-release/chunk_size_100000> ../hyriseBenchmarkTPCH -s 100000`.

**Pull request merged**